src/sas/system/log.py

```diff
@@ -46,17 +46,7 @@ def config_development(self):
 
     def _disable_debug_from_config(self):
         '''disable DEBUG logs as per user configuration (DEBUG logs disabled by default)'''
-        disable_debug = True
-        custom_config = get_custom_config()
-
-        if hasattr(custom_config, "FILTER_DEBUG_LOGS"):
-            if type(custom_config.FILTER_DEBUG_LOGS) is bool:
-                disable_debug = custom_config.FILTER_DEBUG_LOGS
-            else:
-                logging.warning("FILTER_DEBUG_LOGS has invalid value in custom_config.py")
-
-        if disable_debug:
-            # logging.info("Note: DEBUG logs are disabled.")
+        if config.FILTER_DEBUG_LOGS:
```

Comment:
Take it this was the circular reference you were referring to.

I'm using logging in config to give a warning message, one that should probably be fixed by the programmer. Will look into alternatives. I might just use warnings.

Reply:
I've reviewed it, I think its fine.

Reply:
`config_meta.load_from_file_object` (which should probably be `_load_from_file_object`) can call `logger.warning`. With `config.load()` called in `sas.__init__` this happens before `SetupLogger` is called from `MainWindow.run_sasview`.

You can replace the `logger.warning` with `warnings.warn` but that doesn't do any good since I suspect the windows GUI startup eats everything written to stdout without showing it to the user (and I'm guessing many mac users don't know that this info is available through console.app; no idea what the various linux window managers do with it). You do get the warning on save since you have suppressed writing the config file on major version change* so you can just drop the warning here. Or maybe issue a warning after the logger is set up. Note that you should also log the path to the config.json that was loaded, or a list of paths that was searched when it isn't found. It'll make it easier to debug when users complain that their configuration changes were not being saved because we can't find it on load.

I still recommend that you remove the reference from config within the logger so that the order of `config.load` and `SetupLogger` doesn't matter.

Regarding the `config.load` in `sas.__init__`, I have a strong preference for not doing any work when importing, especially for high level imports. I would not expect `import sas` to create a `~/.sasview` directory or a `~/SasView.log` file for example. It looks like the current code does not do so, but it has in the past and I can imagine something like that slipping in later. Also, if you restrict the import to parsing and trivial initialization you don't have to deal with unrelated exceptions on import.

* The "disabled write" warning should tell users to delete the old config file to make the message go away, with full path so they can figure out where cwd is pointing. Another problem with config.json being written to cwd is that this might be some system directory where the user has no write permission. If the current working directory changes when running the program (through a Qt file dialog maybe?) then the saved config.json will not be in the location where load config.json is looking for it.

Reply:
>I still recommend that you remove the reference from config within the logger so that the order of config.load and SetupLogger doesn't matter.

I'm happy with the logger not doing all the weird config stuff that it currently does, that's a legacy thing that I preserved. In my opinion, if someone wants to change the logging behaviour it should really be done through the `.ini`. I just considered this change beyond the scope of this PR. I'll change it.

>  Another problem with config.json being written to cwd is that this might be some system directory where the user has no write permission. If the current working directory changes when running the program (through a Qt file dialog maybe?) then the saved config.json will not be in the location where load config.json is looking for it.

I'd just written it locally for the time being, will move it to `.sasview/config-<major version>.json`

Reply:
Okay, I see that this is a pre-existing issue. I was only looking at the code you added, not the code that you deleted.

Given that there is no way to adjust the `FILTER_DEBUG_LOGS` flag in the GUI or command line and you have said that the user is not supposed to edit `config.json` I think it is safe to disable reference to the config parameter and do the default action `logging.disable(logging.DEBUG)`.

Reply:
Done.